File: memory-domain/src/main/java/com/memory/domain/calendar/repository/RelationshipEventRepositoryCustom.java
package com.memory.domain.calendar.repository;

import com.memory.domain.calendar.RelationshipEvent;
import com.memory.domain.member.Member;

import java.time.LocalDateTime;
import java.util.List;

public interface RelationshipEventRepositoryCustom {

    List<RelationshipEvent> findByMemberAndStartDateTimeBetween(Member member, LocalDateTime startDate, LocalDateTime endDate);

    List<RelationshipEvent> findByMemberAndFutureEvents(Member member);
}

File: memory-domain/src/main/kotlin/com/memory/domain/calendar/repository/RelationshipEventRepositoryCustom.kt
package com.memory.domain.calendar.repository

import com.memory.domain.calendar.RelationshipEvent
import com.memory.domain.member.Member
import java.time.LocalDateTime

interface RelationshipEventRepositoryCustom {

    fun findByMemberAndStartDateTimeBetween(member: Member?, startDate: LocalDateTime?, endDate: LocalDateTime?): List<RelationshipEvent>

    fun findByMemberAndFutureEvents(member: Member?): List<RelationshipEvent>
}
